deploy/runtools/firesim_topology_elements.py

```diff
@@ -336,55 +382,66 @@ def copy_back_job_results_from_run(self, slotno):
                 rootLogger.debug(rsync_cap)
                 rootLogger.debug(rsync_cap.stderr)
 
-    def get_sim_kill_command(self, slotno):
+    def get_sim_kill_command(self, slotno: int) -> str:
         """ return the command to kill the simulation. assumes it will be
         called in a directory where its required_files are already located.
         """
+        assert self.server_hardware_config is not None and isinstance(self.server_hardware_config, RuntimeHWConfig)
         return self.server_hardware_config.get_kill_simulation_command()
 
-    def get_required_files_local_paths(self):
+    def get_required_files_local_paths(self) -> List[Tuple[str, str]]:
         """ Return local paths of all stuff needed to run this simulation as
         an array. """
         all_paths = []
 
-        if self.get_job().rootfs_path() is not None:
-            all_paths.append([self.get_job().rootfs_path(), self.get_rootfs_name()])
+        job_rootfs_path = self.get_job().rootfs_path()
```

Comment:
IDK why but this was surprisingly hard to get `mypy` to cooperate hence why this is verbose now.